app/assets/javascripts/categories_legend.js

```diff
@@ -1,4 +1,8 @@
 var CategoryLegend = {
+  color_map: {},
+  selected_values: [],
```

Comment:
it's already stored in `params['q['tags_filter']]`. Use getter/setter on one of this place.